subsys/net/l2/openthread/Kconfig

```diff
@@ -224,10 +224,39 @@ config OPENTHREAD_SHELL
 	select SHELL
 	default y
 
-config OPENTHREAD_DIAG
-	bool "Diagnostic functions support"
+config OPENTHREAD_BACKBONE_ROUTER
```

Comment:
Given we're doing a cleanup here, I think it would make sense to put the stack configuration options into a separate submenu (i. e. those options that translate into OT ones). Currently the OT menu in Kconfig is quite bulky, and hard to differentiate what's actually changing the stack configuration and what affecting the OpenThread L2. Optionally, they could also be put into a separate Kconfig file (like `Kconfig.openthread`) and sourced here.

Reply:
Applied changes as suggested.